hxtool/cli/poke.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+from argparse import ArgumentError, ArgumentTypeError
+from binascii import hexlify, unhexlify
+from logging import getLogger
+import os
+import re
+
+import hxtool
+from .base import CliCommand
+
+logger = getLogger(__name__)
+
+
+def default_length():
+    try:
+        length = os.environ.get("HXTOOL_PEEK_LENGTH", "1")
+        return hexadecimal_number(length)
+    except (ArgumentTypeError, ValueError):
+        return 1
+
+
+def hexadecimal_number(string):
+    number = int(string, 16)
+    if number < 0:
+        raise ArgumentTypeError("must be positive")
+    return number
+
+
+def hexadecimal_data(string):
+    return unhexlify(bytes(re.sub(r"\A0x", "", string), 'utf-8'))
+
+
+class PokeCommand(CliCommand):
+
+    name = "poke"
+    help = "Read/write hex bytes in the device memory (advanced feature, use caution)"
+
+    @staticmethod
+    def setup_args(parser) -> None:
```

Comment:
I am not happy with the argument logic here. If you specify `--data`, then what is  `--length` supposed to do? I suggest the following:

`hxtool poke --start 0x100 --data 1234` with both args mandatory

and

`hxtool peek --start 0x100 --end 0x120` or
`hxtool peek --start 0x100 --lenght 0x20`, so mandatory start and either end or length.